Sources/Datadog/RUM/RUMMonitor/Scopes/RUMApplicationScope.swift

```diff
@@ -34,6 +48,24 @@ internal class RUMApplicationScope: RUMScope {
     let context: RUMContext
 
     func process(command: RUMCommand) -> Bool {
+        if let currentSession = sessionScope {
+            let shouldRefreshSession = currentSession.process(command: command)
+            if shouldRefreshSession {
+                let refreshedSession = RUMSessionScope(parent: self, dependencies: dependencies)
+                sessionScope = refreshedSession
+                _ = refreshedSession.process(command: command)
+            }
+        } else {
+            switch command {
+            case .startView:
```

Comment:
i didn't understand the logic here: if a session is expired, new one can start with any command; however if there is no expired session, new session can only start with `startView`. 
why are those 2 cases different?

Reply:
It's coming from the first product requirement:

> The RUM Session monitoring has 3 requirements:
> * the initial Session is started with the first View,
> * ...
> * ...

So the session starts as soon as some view is displayed in the app. Once the session is started, RUM always has a valid session. Although, it doesn't need to be the same session, because we implement session timeout and expiration logic.